src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements FigureInformation {
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private String color = colorSupplier.getRandomColor();
+
+    public abstract Figure createRandomFigure();
+
+    protected int getRandomInt() {
+        return random.nextInt(100);
+    }
```

Comment:
You can pass random values to the constructor of a figure instead of generating them inside figure classes. This method should be moved to the `FigureSupplier` class.